### PR TITLE
Use Docker API bindings in kibana_proxy.py (#1891)

### PR DESCRIPTION
**Pull request by natanlao**
Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `HCA` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] Added `chain` label <sub>or this PR is not the base of another PR</sub>
- [x] Made this PR a blocker of next PR in chain <sub>or this PR is not the base of another PR</sub>
- [x] Ran `make update_requirements` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `[r]` prefix of commit title
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox` <sub>or commented that sandbox will be skipped</sub>
- [x] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit

Primary reviewer (after pushing merge commit)

- [ ] Moved linked issue to Merged
- [ ] Pushed merge commit to Github
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Unassign reviewer from PR

### CODE REVIEW

File: scripts/kibana-proxy.py [hunk 1]
```diff
@@ -48,46 +39,60 @@ class KibanaProxy:
...
+                                            detach=True,
+                                            network_mode=f'container:{proxy.name}',
+                                            tty=True)
+            self.containers.append(cerebro)
             time.sleep(10)
             print(f'Now open Kibana at http://127.0.0.1:{kibana_port}/')
-            print(f'Now open Cerebro at http://127.0.0.1:{cerebro_port}/ and paste http://localhost:{proxy_port}')
-            self.wait()
+            print(f'Now open Cerebro at http://127.0.0.1:{cerebro_port}/ '
+                  f'and paste http://localhost:{proxy_port}')
```
Comment (noah-aviel-dove) [inline scripts/kibana-proxy.py:88]: Be aware that you can avoid the manual/copy paste using
```suggestion
            print(f'Now open Cerebro at http://127.0.0.1:{cerebro_port}/#/overview?host=http://localhost:{proxy_port}')
```
But there might be some other use-case I'm not aware of where the user would want to specify a different host.

Comment (natanlao) [reply scripts/kibana-proxy.py:88]: This is great, thanks! I was looking for something like this but wasn't able to find it.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
It bugged me that the container output was suppressed. Please review and test my changes (I tested them, too).

The respective section in the README needs updating, too.

Could you also rename kibana-proxy.pl to `kibana_proxy.pl`? You should do that as a split commit (first rename, then update references) since the script refers to itself.

**Comment by natanlao** [comment]
@hannes-ucsc 

> You should do that as a split commit (first rename, then update references) since the script refers to itself.

Out of curiosity, is there a functional difference between this approach and doing the work in a single commit, or is this something we do out of clarity/preference?

**Comment by hannes-ucsc** [comment]
> Out of curiosity, is there a functional difference between this approach and doing the work in a single commit, or is this something we do for sake of clarity / out of preference?

In my experience, Github's heuristic for detecting file moves fails when renaming AND modifying a file. If that happens, the change shows up as a deletion and an addition and the change history of the lines in the file is lost.

**Review by hannes-ucsc — APPROVED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
For demo, just demonstrate the use of the script. Show that containers are cleaned-up on exit.

**Pull request merged**